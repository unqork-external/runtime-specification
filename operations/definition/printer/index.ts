export * from './printer.model'
export * from './printer.options'
