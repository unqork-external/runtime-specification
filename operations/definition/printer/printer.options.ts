import { Default, Property } from '@tsed/schema'

import { BaseOperationOptions } from '../../interface'

export class PrinterOperationOptions extends BaseOperationOptions {
  @Property()
  thingToPrint: string

  @Property()
  @Default('log')
  logType: string = 'log'
}
