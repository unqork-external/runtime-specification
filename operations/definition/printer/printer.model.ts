import { DiscriminatorValue, Required } from '@tsed/schema'

import { PrinterOperationOptions } from './printer.options'
import { Alpha } from '../../../../decorators'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.PRINTER)
@Alpha()
export class PrinterOperation extends Operation<OperationTypes.PRINTER> {
  public type = OperationTypes.PRINTER as const

  @Required()
  public options: PrinterOperationOptions = new PrinterOperationOptions()
}
