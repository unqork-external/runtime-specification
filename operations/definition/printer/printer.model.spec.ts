import { PrinterOperation } from './printer.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: Printer Tests', () => {
  it('Printer Model', () => {
    const { schema } = generateSchemaAndValidate(PrinterOperation)

    expect(schema.properties).toEqual({
      type: { type: 'string', const: 'PRINTER', examples: ['PRINTER'] },
      name: { type: 'string', description: 'Name of the operation' },
      options: { $ref: '#/definitions/PrinterOperationOptions' },
    })
  })
})
