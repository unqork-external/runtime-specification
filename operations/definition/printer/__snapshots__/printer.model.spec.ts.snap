// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Operation Model: Printer Tests Printer Model 1`] = `
{
  "autogenDisabled": true,
  "definitions": {
    "PrinterOperationOptions": {
      "properties": {
        "logType": {
          "default": "log",
          "type": "string",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "thingToPrint": {
          "type": "string",
        },
      },
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/PrinterOperationOptions",
    },
    "type": {
      "const": "PRINTER",
      "examples": [
        "PRINTER",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
